### ISSUE TITLE
Creating config file without WIFI

### ISSUE THREAD

**Issue by honeybadger2023** [bdring/fluidterm#6]
Hi, 

I'm trying to write a config file without the use of WIFI or any Serial communication. Fluidterm is not working. It doesn't connect. So I decided to "hardcode" a config file by using the FluidPath and Filestream classes like it is done within the WebUI but I keep failing.

What is the best way to get a config file on the board without any external interfaces?

Is it possible to just use littlefs or spiffs to write the file?

**Comment by honeybadger2023** [comment]
And another question: I use a custom ESP motion controller PCB and one Stepper is connected to a PIN that FluidNC would like to use for uart. Is it possible to completely disable any UART or is this needed?

**Comment by MitchBradley** [comment]
If you can compile the firmware, you can put your config file in the data/ directory and execute "pio run -t buildfs" to create a filesystem image that you can upload with esptool.  It will be located in .pio/build/wifi/littlefs.bin

But you really should take a step back and get FluidTerm working.  Maintenance will be very difficult otherwise.

If you want to disable all UARTs, you will have to modify the code.  We don't have time to help with custom boards.  All of our normal use cases assume that a UART is available, at least for maintenance.

**Comment by honeybadger2023** [comment]
Ok. Thanks. Is the I2S and Uart in some way critical for the process if RMT is used?

**Comment by MitchBradley** [comment]
I2s and rmt are mutually exclusive.  i2s is only for systems with shift-register stepping interface.  UART is for communication with sender and debugging.

**Comment by honeybadger2023** [comment]
So it is possible to completely remove all I2S based stuff from the code without rendering it useless?

**Comment by MitchBradley** [comment]
Yes, but as I said, I don't have time to support mods like this.  You will just have to grapple with it.

**Comment by honeybadger2023** [comment]
I totally understand and thanks for you help.
